var/spack/repos/builtin/packages/py-mkdocs/package.py

```diff
@@ -14,17 +14,26 @@ class PyMkdocs(PythonPackage):
     homepage = "https://www.mkdocs.org/"
     pypi = "mkdocs/mkdocs-1.3.1.tar.gz"
 
+    version("1.4.2", sha256="8947af423a6d0facf41ea1195b8e1e8c85ad94ac95ae307fe11232e0424b11c5")
     version("1.3.1", sha256="a41a2ff25ce3bbacc953f9844ba07d106233cd76c88bac1f59cb1564ac0d87ed")
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"), when="@1.4.2:")
     depends_on("py-setuptools", type="build")
+    depends_on("py-hatchling", type="build", when="@1.4.2:")
     depends_on("py-click@3.3:", type=("build", "run"))
+    depends_on("py-click@7.0:", type=("build", "run"), when="@1.4.2:")
     depends_on("py-jinja2@2.10.2:", type=("build", "run"))
+    depends_on("py-jinja2@2.11.1:", type=("build", "run"), when="@1.4.2:")
     depends_on("py-markdown@3.2.1:3.3", type=("build", "run"))
+    depends_on("py-markdown@3.2.1:3.4", type=("build", "run"), when="@1.4.2:")
     depends_on("py-pyyaml@3.10:", type=("build", "run"))
+    depends_on("py-pyyaml@5.1:", type=("build", "run"), when="@1.4.2:")
     depends_on("py-watchdog@2.0:", type=("build", "run"))
     depends_on("py-ghp-import@1.0:", type=("build", "run"))
     depends_on("py-pyyaml-env-tag@0.1:", type=("build", "run"))
     depends_on("py-importlib-metadata@4.3:", type=("build", "run"))
     depends_on("py-packaging@20.5:", type=("build", "run"))
     depends_on("py-mergedeep@1.3.4:", type=("build", "run"))
+    depends_on("py-colorama@0.4:", type=("build", "run"), when="platform=win32")
+    depends_on("py-babel@2.9.0:", type=("build", "run"), when="@1.4.2:")
```

Comment:
This is optional, I would either remove it or add a variant.

Reply:
That's what I thought too, but when I tried to test the install without babel (since it was listed as optional) it would throw an error saying that something with babel couldn't be located and would fail.

Reply:
How should I go forward with this? Should I try and figure out why the optional dependency is required or should I leave it like this for now so it can actually build?

Reply:
I updated the package file on my branch to this new version and then ran `spack install py-mkdocs@1.4.2`. This was when I didn't include babel because I first wanted to make sure that the new required dependencies would build and function, then I was going to add a variant for babel.

Reply:
So `spack install py-mkdocs` failed to build?

Reply:
Correct. When I looked at the build log, it gave a module not found error for babel:
```
Traceback (most recent call last):
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-21.3.1-4urh6qgy5mmysevxpqmae57lbjiaekbw/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
      main()
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-21.3.1-4urh6qgy5mmysevxpqmae57lbjiaekbw/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-21.3.1-4urh6qgy5mmysevxpqmae57lbjiaekbw/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 261, in build_wheel
      return _build_backend().build_wheel(wheel_directory, config_settings,
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-hatchling-1.10.0-ot2cbjhvcmlqimsb2mbc2uugvphfccs4/lib/python3.9/site-packages/hatchling/build.py", line 59, in build_wheel
      return os.path.basename(next(builder.build(wheel_directory, ['standard'])))
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-hatchling-1.10.0-ot2cbjhvcmlqimsb2mbc2uugvphfccs4/lib/python3.9/site-packages/hatchling/builders/plugin/interface.py", line 136, in build
      build_hook.initialize(version, build_data)
    File "/home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-rerdvbtrxwub652yljysy3ydxt5ikx2v/spack-src/hatch_build.py", line 8, in initialize
      from babel.messages.frontend import compile_catalog
  ModuleNotFoundError: No module named 'babel'
```

Reply:
Can you upload the build log here?

Reply:
Yeah, here's the whole thing. This is when I ran the most recent commit but commented out the babel dependency.
```
==> py-mkdocs: Executing phase: 'install'
==> [2022-11-15-10:47:30.745889] '/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/python-3.10.8-apcozfjhhvdn334ixipxfsjzzczwlvgg/bin/python3.10' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql' '.'
Using pip 22.2.2 from /home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip (python 3.10)
Non-user install due to --prefix or --target option
Ignoring indexes: https://pypi.org/simple
Created temporary directory: /tmp/pip-ephem-wheel-cache-yhll3v7z
Created temporary directory: /tmp/pip-build-tracker-xb2fdbbw
Initialized build tracking at /tmp/pip-build-tracker-xb2fdbbw
Created build tracker: /tmp/pip-build-tracker-xb2fdbbw
Entered build tracker: /tmp/pip-build-tracker-xb2fdbbw
Created temporary directory: /tmp/pip-install-ia1z30qq
Processing /home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql/spack-src
  Added file:///home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql/spack-src to build tracker '/tmp/pip-build-tracker-xb2fdbbw'
  Created temporary directory: /tmp/pip-modern-metadata-x82hxg28
  Preparing metadata (pyproject.toml): started
  Running command Preparing metadata (pyproject.toml)
  Preparing metadata (pyproject.toml): finished with status 'done'
  Source in /home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql/spack-src has version 1.4.2, which satisfies requirement mkdocs==1.4.2 from file:///home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql/spack-src
  Removed mkdocs==1.4.2 from file:///home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql/spack-src from build tracker '/tmp/pip-build-tracker-xb2fdbbw'
Created temporary directory: /tmp/pip-unpack-rjq74uci
Building wheels for collected packages: mkdocs
  Created temporary directory: /tmp/pip-wheel-kj5n1job
  Destination directory: /tmp/pip-wheel-kj5n1job
  Building wheel for mkdocs (pyproject.toml): started
  Running command Building wheel for mkdocs (pyproject.toml)
  Traceback (most recent call last):
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
      main()
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 261, in build_wheel
      return _build_backend().build_wheel(wheel_directory, config_settings,
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-hatchling-1.10.0-xs6bptr5oy7fn3reb75zcw6kjjqwm2o7/lib/python3.10/site-packages/hatchling/build.py", line 59, in build_wheel
      return os.path.basename(next(builder.build(wheel_directory, ['standard'])))
    File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-hatchling-1.10.0-xs6bptr5oy7fn3reb75zcw6kjjqwm2o7/lib/python3.10/site-packages/hatchling/builders/plugin/interface.py", line 136, in build
      build_hook.initialize(version, build_data)
    File "/home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql/spack-src/hatch_build.py", line 8, in initialize
      from babel.messages.frontend import compile_catalog
  ModuleNotFoundError: No module named 'babel'
  error: subprocess-exited-with-error
  
  × Building wheel for mkdocs (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/python-3.10.8-apcozfjhhvdn334ixipxfsjzzczwlvgg/bin/python3.10 /home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpnp8f0bsl
  cwd: /home/eshe/spack/etc/spack/eshe/tmp/spack-stage/spack-stage-py-mkdocs-1.4.2-ju6trzv65ozlnxzuwiijcs3teuepn6ql/spack-src
  Building wheel for mkdocs (pyproject.toml): finished with status 'error'
  ERROR: Failed building wheel for mkdocs
Failed to build mkdocs
ERROR: Could not build wheels for mkdocs, which is required to install pyproject.toml-based projects
Exception information:
Traceback (most recent call last):
  File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip/_internal/cli/base_command.py", line 167, in exc_logging_wrapper
    status = run_func(*args)
  File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip/_internal/cli/req_command.py", line 247, in wrapper
    return func(self, options, args)
  File "/home/eshe/spack/opt/spack/linux-rhel9-icelake/gcc-12.1.0/py-pip-22.2.2-d66px6i3tgm6fxvh4hqurcm5tib56pn7/lib/python3.10/site-packages/pip/_internal/commands/install.py", line 431, in run
    raise InstallationError(
pip._internal.exceptions.InstallationError: Could not build wheels for mkdocs, which is required to install pyproject.toml-based projects
Removed build tracker: '/tmp/pip-build-tracker-xb2fdbbw'
```

Reply:
Can you add a comment to this effect? Something like "This dep is not optional, build fails without it". This will help me a year from now when someone tries to update this package and I assume the dep is optional.

Reply:
Sure, will do.